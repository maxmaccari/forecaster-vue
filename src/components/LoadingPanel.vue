<template>
  <VPanel class="max-w-64" :separator="false" title-orientation="column">
    <template #title>
      <VLoadingSpinner class="block w-24 m-auto" />

      <span class="w-full">Loading...</span>
    </template>
  </VPanel>
</template>

<script lang="ts" setup>
  import VLoadingSpinner from './VLoadingSpinner.vue'
  import VPanel from './VPanel.vue';
</script>
