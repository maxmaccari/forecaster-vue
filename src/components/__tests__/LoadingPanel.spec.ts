import { describe, it, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils'
import LoadingPanel from '../LoadingPanel.vue'

describe('LoadingPanel', () => {
  it('renders the component properly', () => {
    const wrapper = shallowMount(LoadingPanel)

    expect(wrapper.html()).toMatchSnapshot()
  })
})
