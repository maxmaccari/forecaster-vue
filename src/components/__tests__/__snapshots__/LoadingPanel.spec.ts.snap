// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingPanel > renders the component properly 1`] = `"<v-panel-stub separator="false" titleorientation="column" bgcolor="bg-secondary/80" class="max-w-64"></v-panel-stub>"`;
